core/src/main/java/org/elasticsearch/index/query/WrapperQueryBuilder.java

```diff
@@ -86,4 +92,44 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     public String getWriteableName() {
         return NAME;
     }
+
+    @Override
+    protected Query doToQuery(QueryShardContext context) throws IOException {
+        try (XContentParser qSourceParser = XContentFactory.xContent(source).createParser(source)) {
+            final QueryParseContext parseContext = new QueryParseContext(context);
+            parseContext.reset(qSourceParser);
+            Query query = parseContext.parseInnerQuery();
+            context.combineNamedQueries(context);  // NO COMMIT: not really sure if this is the right way
```

Comment:
I had a look at this, I think we can do the following:

```
QueryBuilder queryBuilder = parseContext.parseInnerQueryBuilder();
 return queryBuilder.toQuery(context);
```

Given that we reuse the original query shard context, I think we my not need to call combineNamedQueries, actually we could remove that method cause this is the only reason why we have it in the first place. What do you think @cbuescher ?